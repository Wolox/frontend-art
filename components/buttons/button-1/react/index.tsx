import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
