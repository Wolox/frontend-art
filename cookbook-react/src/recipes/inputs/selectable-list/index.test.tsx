import React from 'react';
import { render } from '@testing-library/react';

import SelectableListExample from '.';

test('Should match snapshot', () => {
  const { container } = render(<SelectableListExample />);
  expect(container).toMatchSnapshot();
});
