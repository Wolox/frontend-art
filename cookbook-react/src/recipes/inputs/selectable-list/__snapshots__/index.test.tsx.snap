// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <div
    class="exampleContainer"
  >
    <span
      class="listTitle"
    >
      With input as children
    </span>
    <ul
      class="selectableListContainer"
    >
      <li
        class="item"
      >
        <div
          class="item"
        >
          <input
            class="input"
            id="list-1-1"
            name="list-1-first_item"
            type="radio"
          />
          <label
            for="list-1-1"
          >
            First item
          </label>
        </div>
      </li>
      <li
        class="item"
      >
        <div
          class="item"
        >
          <input
            class="input"
            id="list-1-2"
            name="list-1-second_item"
            type="radio"
          />
          <label
            for="list-1-2"
          >
            Second item
          </label>
        </div>
      </li>
      <li
        class="item"
      >
        <div
          class="item"
        >
          <input
            class="input"
            id="list-1-3"
            name="list-1-third_item"
            type="radio"
          />
          <label
            for="list-1-3"
          >
            Third item
          </label>
        </div>
      </li>
    </ul>
    <span
      class="listTitle"
    >
      With a custom component as children and multiple selection
    </span>
    <ul
      class="selectableListContainer"
    >
      <li
        class="item item"
      >
        <input
          class="input"
          id="list2-1"
          name="list-2-first_item"
          type="checkbox"
        />
        <label
          for="list2-1"
        >
          First item
        </label>
      </li>
      <li
        class="item item"
      >
        <input
          class="input"
          id="list2-2"
          name="list-2-second_item"
          type="checkbox"
        />
        <label
          for="list2-2"
        >
          Second item
        </label>
      </li>
      <li
        class="item item"
      >
        <input
          class="input"
          id="list2-3"
          name="list-2-third_item"
          type="checkbox"
        />
        <label
          for="list2-3"
        >
          Third item
        </label>
      </li>
    </ul>
    <span
      class="listTitle"
    >
      With multiple selection and initialValues
    </span>
    <ul
      class="selectableListContainer"
    >
      <li
        class="item item"
      >
        <div
          class="item"
        >
          <input
            checked=""
            class="input"
            id="list-3-1"
            name="list-3-first_item"
            type="checkbox"
          />
          <label
            for="list-3-1"
          >
            First item
          </label>
        </div>
      </li>
      <li
        class="item item"
      >
        <div
          class="item"
        >
          <input
            checked=""
            class="input"
            id="list-3-2"
            name="list-3-second_item"
            type="checkbox"
          />
          <label
            for="list-3-2"
          >
            Second item
          </label>
        </div>
      </li>
      <li
        class="item item"
      >
        <div
          class="item"
        >
          <input
            class="input"
            id="list-3-3"
            name="list-3-third_item"
            type="checkbox"
          />
          <label
            for="list-3-3"
          >
            Third item
          </label>
        </div>
      </li>
    </ul>
    <span
      class="listTitle"
    >
      With no multiple selection and initialValue
    </span>
    <ul
      class="selectableListContainer"
    >
      <li
        class="item item"
      >
        <div
          class="item"
        >
          <input
            class="input"
            id="list-4-1"
            name="list-4-first_item"
            type="radio"
          />
          <label
            for="list-4-1"
          >
            First item
          </label>
        </div>
      </li>
      <li
        class="item item"
      >
        <div
          class="item"
        >
          <input
            checked=""
            class="input"
            id="list-4-2"
            name="list-4-second_item"
            type="radio"
          />
          <label
            for="list-4-2"
          >
            Second item
          </label>
        </div>
      </li>
      <li
        class="item item"
      >
        <div
          class="item"
        >
          <input
            class="input"
            id="list-4-3"
            name="list-4-third_item"
            type="radio"
          />
          <label
            for="list-4-3"
          >
            Third item
          </label>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
